pkg/algorithm/hived_algorithm.go

```diff
@@ -1344,7 +1344,8 @@ func (h *HivedAlgorithm) releaseGpu(pGpu *PhysicalCell, vcn api.VirtualClusterNa
 // from the free cell list, and track the cell usages for safety and doomed bad cell check.
 func (h *HivedAlgorithm) allocatePreassignedCell(
 	c *PhysicalCell,
-	vcn api.VirtualClusterName) (safetyOk bool, reason string) {
+	vcn api.VirtualClusterName,
+	doomedBad bool) (safetyOk bool, reason string) {
 
```

Comment:
Consider to always update our VC doomed state, if any "badFreeCells, totalLeftCellNum, vcFreeNum" changed,